zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/ElasticsearchSpanConsumer.java

```diff
@@ -119,13 +121,22 @@ void add(long indexTimestamp, Span span, long timestampMillis) {
         writer.name("_q");
         writer.beginArray();
         for (Annotation a : span.annotations()) {
-          if (a.value().length() > 255) continue;
+          if (a.value().length() > INDEX_CHARS_LIMIT) {
+            continue;
+          }
           writer.value(a.value());
         }
         for (Map.Entry<String, String> tag : span.tags().entrySet()) {
-          if (tag.getKey().length() + tag.getValue().length() + 1 > 255) continue;
+          int length = tag.getKey().length() + tag.getValue().length() + 1;
+          if (length > INDEX_CHARS_LIMIT) {
+            continue;
+          }
           writer.value(tag.getKey()); // search is possible by key alone
-          writer.value(tag.getKey() + "=" + tag.getValue());
+          writer.value(
+            new StringBuilder(length)
```

Comment:
I'm pretty sure javac does this for us :P

Reply:
oic yeah suppose it wouldn't pre-define the length of the builder, though the length here is only correct if ascii (utf 8 will overrun this)

Reply:
though maybe ES length is chars anyway? regardless, that's a different issue.. I'm fine with this.